doc/contribute/guidelines.rst

```diff
@@ -374,6 +374,19 @@ Note, gitlint only checks HEAD (the most recent commit), so you should run it
 after each commit, or use the ``--commits`` option to specify a commit range
 covering all the development patches to be submitted.
 
+yamllint
```

Comment:
This is run by the check_compliance.py script.  I'm not sure we need this.  

Alternately, the info about yamllint probably belongs under the Coding Style section instead of here with the git setup.

Reply:
But then shouldn't we also remove the mention to how to manually run gitlint (in the previous paragraph), if the recommendation is to use `check_compliance.py` as the go-to place for checking a PR is compliant?

Reply:
That's a good point.  I only see one passing reference to `check_compliance.py` in the documentation.  This is an oversight in my opinion. 

At a minimum, I think we should add basic instructions for running `check_compliance.py` as part of the Contribution Workflow.

Looking at the outline of this doc page, it could use some updates.

How about creating a new section "PR Guidelines" with the following subsections.
```
Git Setup # note - drops gitlint, twister, and clang-format info
Pull Request Guidelines
    Commit Message Guidelines
    Coding Style
        clang-format
    Continuous Integration
    Running CI Tests Locally
        check_compliance.py
        twister
```

Reply:
Also, @nashif  generated this top 10 list  a couple years ago.  There is some good content that would nice to have the documentation:

 Here is a TOP10 list to improve and accelerate the review process of pull requests. If you follow them, chances are your pull request will get the attention needed and it will be ready for merge sooner than later:
1. Make sure you are on top of latest main branch
    * git fetch upstream
    * git rebase upstream/main
1. Run compliance tests locally and fix any issues before submitting a PR
    * ./scripts/ci/check_compliance.py -c upstream/main..
1. Make sure title of PR explains what is being fixed or added
1. Make sure your PR has a body with more details about the content of your submission
1. Make sure you reference the issue you are fixing in the body of the PR
1. Add reviewers if you can and know who will be able to review your code
1. Set the labels if those are not automatically set by the bot.
1. Watch early CI results immediately after submissions and fix issues as they are discovered
1. Revisit PR after 1-2 hours to see the status of all CI checks, make sure all is green
1. If you get request for changes and submit a change to address them, make sure you re-request review from those who asked for the changes

Reply:
Appreciate your feedback. I'll update this PR addressing the suggested changes.